src/components/Comments/ReplyCommentItem.js

```diff
@@ -141,6 +159,61 @@ const ReplyCommentItem = ({
     }
   };
 
+  const username = comment?.data?.anon_user_info_color_name
+    ? `Anonymous ${comment.data?.anon_user_info_emoji_name}`
+    : user?.data?.username;
+
+  const onPressDM = async () => {
+    try {
+      setLoading({...loading, loadingDm: true});
+      if (!comment?.data?.anon_user_info_color_name) {
+        const channelName = [username, profile?.myProfile?.username].join(',');
+        const selectedUser = {
+          user: {
+            name: channelName,
+            image: comment?.user?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+          }
+        };
+        const members = [comment?.user?.id, profile.myProfile.user_id];
+        await createSignChat(members, selectedUser);
+      } else {
+        await sendMessageDM(comment?.id, 'comment', 'SIGNED');
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDm: false});
+    }
+  };
+
+  const onPressDMAnon = async () => {
+    try {
+      setLoading({...loading, loadingDmAnon: true});
+      await sendMessageDM(comment?.id, 'comment', 'ANONYMOUS');
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDmAnon: false});
+    }
+  };
+
+  const dataSheet = [
+    {
+      id: 1,
+      name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+      icon: <SendDMBlack />,
+      onPress: onPressDM
+    },
+    {
+      id: 2,
+      name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
+      icon: <SendDMAnonBlock />,
+      onPress: onPressDMAnon
+    }
+  ];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are a couple of issues here:

1. Error handling: The catch block in the `onPressDM` and `onPressDMAnon` functions only logs the error to the console. It would be better to display an error message to the user or handle it in some other way.

2. Loading state: The loading state is being updated multiple times within the `onPressDM` and `onPressDMAnon` functions. This could potentially cause unnecessary re-renders. Consider using a separate state for each loading action or updating the loading state once at the end of the function.

3. Direct mutation of state: In the `onPressDM` and `onPressDMAnon` functions, the loading state is directly mutated which is not recommended in React. Instead, use functional updates when the new state depends on the previous one.

Here's how you can address these issues:

```diff
-   const onPressDM = async () => {
-     try {
-       setLoading({...loading, loadingDm: true});
-       if (!comment?.data?.anon_user_info_color_name) {
-         const channelName = [username, profile?.myProfile?.username].join(',');
-         const selectedUser = {
-           user: {
-             name: channelName,
-             image: comment?.user?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
-           }
-         };
-         const members = [comment?.user?.id, profile.myProfile.user_id];
-         await createSignChat(members, selectedUser);
-       } else {
-         await sendMessageDM(comment?.id, 'comment', 'SIGNED');
-       }
-     } catch (e) {
-       console.warn(e);
-     } finally {
-       refSheet.current.close();
-       setLoading({...loading, loadingDm: false});
-     }
-   };
+   const onPressDM = async () => {
+     setLoading(prevLoading => ({...prevLoading, loadingDm: true}));
+     try {
+       if (!comment?.data?.anon_user_info_color_name) {
+         const channelName = [username, profile?.myProfile?.username].join(',');
+         const selectedUser = {
+           user: {
+             name: channelName,
+             image: comment?.user?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+           }
+         };
+         const members = [comment?.user?.id, profile.myProfile.user_id];
+         await createSignChat(members, selectedUser);
+       } else {
+         await sendMessageDM(comment?.id, 'comment', 'SIGNED');
+       }
+     } catch (e) {
+       // handle error here
+     } finally {
+       refSheet.current.close();
+       setLoading(prevLoading => ({...prevLoading, loadingDm: false}));
+     }
+   };

-   const onPressDMAnon = async () => {
-     try {
-       setLoading({...loading, loadingDmAnon: true});
-       await sendMessageDM(comment?.id, 'comment', 'ANONYMOUS');
-     } catch (e) {
-       console.warn(e);
-     } finally {
-       refSheet.current.close();
-       setLoading({...loading, loadingDmAnon: false});
-     }
-   };
+   const onPressDMAnon = async () => {
+     setLoading(prevLoading => ({...prevLoading, loadingDmAnon: true}));
+     try {
+       await sendMessageDM(comment?.id, 'comment', 'ANONYMOUS');
+     } catch (e) {
+       // handle error here
+     } finally {
+       refSheet.current.close();
+       setLoading(prevLoading => ({...prevLoading, loadingDmAnon: false}));
+     }
+   };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->